synapse/telepath.py

```diff
@@ -374,6 +374,60 @@ def __call__(self, *args, **kwargs):
         todo = (self.name, args, kwargs)
         return GenrIter(self.proxy, todo, self.share)
 
+class Pipeline(s_base.Base):
+
+    async def __anit__(self, proxy, genr, name=None):
+
+        await s_base.Base.__anit__(self)
+
+        self.genr = genr
+        self.name = name
+        self.proxy = proxy
+
+        self.count = 0
+
+        self.link = await proxy.getPoolLink()
```

Comment:
The link we got from the pool never goes back in the event of a clean teardown of the Pipeline?

Reply:
Line 415?